x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/services/analytics_service/delete_analytics.ts

```diff
@@ -39,3 +37,114 @@ export const deleteAnalytics = async (d: DataFrameAnalyticsListRow) => {
   }
   refreshAnalyticsList$.next(REFRESH_ANALYTICS_LIST_STATE.REFRESH);
 };
+
+export const deleteAnalyticsAndTargetIndex = async (
+  d: DataFrameAnalyticsListRow,
+  deleteTargetIndex: boolean,
+  deleteTargetPattern: boolean
+) => {
+  const toastNotifications = getToastNotifications();
+  const destinationIndex = d.config.dest.index;
```

Comment:
Might need to check for an array here - sometimes `dest.index` can be an array, I believe. 
Something like 
```
const destIndex = Array.isArray(d.config.dest.index)
              ? d.config.dest.index[0]
              : d.config.dest.index;
```

Reply:
Done in https://github.com/elastic/kibana/pull/66934/commits/5bd97e9c466ad1a88d32da4f432583d5ddc3aed0